dashboard/src/components/billing/UsageUpgradeBanner.tsx

```diff
@@ -0,0 +1,60 @@
+'use client';
+
+import { AlertTriangle, AlertCircle } from 'lucide-react';
+import { Button } from '@/components/ui/button';
+import Link from 'next/link';
+import { use } from 'react';
+import type { UserBillingData } from '@/entities/billing';
+import { isClientFeatureEnabled } from '@/lib/client-feature-flags';
+
+interface UsageUpgradeBannerProps {
+  billingDataPromise: Promise<UserBillingData>;
+}
+
+export default function UsageUpgradeBanner({ billingDataPromise }: UsageUpgradeBannerProps) {
+  if (!isClientFeatureEnabled('enableBilling')) {
+    return null;
+  }
+
+  const billingData = use(billingDataPromise);
+
+  if (!billingData.usage.isOverLimit) {
+    return null;
+  }
+
+  const overageEvents = billingData.usage.current - billingData.usage.limit;
+  const overagePercentage = ((overageEvents / billingData.usage.limit) * 100).toFixed(1);
+
+  return (
+    <div className='w-full border-b border-red-600 bg-gradient-to-r from-red-500 to-orange-500 text-white shadow-sm'>
+      <div className='container mx-auto px-4 py-2'>
+        <div className='flex items-center justify-between'>
+          <div className='flex items-center gap-3'>
+            <AlertTriangle className='h-4 w-4 flex-shrink-0' />
+            <div className='flex flex-col text-sm sm:flex-row sm:items-center sm:gap-2'>
+              <span className='font-medium'>Usage Limit Exceeded:</span>
+              <span>
+                You&apos;ve used <strong>{billingData.usage.current.toLocaleString()}</strong> events out of your{' '}
+                <strong>{billingData.usage.limit.toLocaleString()}</strong> event limit ({overagePercentage}% over
```

Comment:
Quibble: Reuse your nice formatters 🔥 
```suggestion
                You&apos;ve used <strong>{formatNumber(billingData.usage.current)}</strong> events out of your{' '}
                <strong>{formatNumber(billingData.usage.limit)}</strong> event limit ({formatPercentage(overagePercentage)} over
```